src/test/app/DepositAuth_test.cpp

```diff
@@ -387,7 +378,325 @@ struct DepositAuth_test : public beast::unit_test::suite
+
+            env (pay (gw, alice, USD (1000)));
+            env.close();
+
+            // Make XRP and IOU payments from alice to becky.  Should be fine.
+            env (pay (alice, becky, XRP (100)));
+            env (pay (alice, becky, USD (100)));
+            env.close();
+
+            // becky decides to require authorization for deposits.
+            env(fset (becky, asfDepositAuth));
+            env.close();
+
+            // alice can no longer pay becky.
+            env (pay (alice, becky, XRP (100)), ter (tecNO_PERMISSION));
+            env (pay (alice, becky, USD (100)), ter (tecNO_PERMISSION));
+            env.close();
+
+            // becky preauthorizes carol for deposit, which doesn't provide
+            // authorization for alice.
+            env (deposit::auth (becky, carol));
+            env.close();
+
+            // alice still can't pay becky.
+            env (pay (alice, becky, XRP (100)), ter (tecNO_PERMISSION));
+            env (pay (alice, becky, USD (100)), ter (tecNO_PERMISSION));
+            env.close();
+
+            // becky preauthorizes alice for deposit.
+            env (deposit::auth (becky, alice));
+            env.close();
+
+            // alice can now pay becky.
+            env (pay (alice, becky, XRP (100)));
+            env (pay (alice, becky, USD (100)));
+            env.close();
+
+            // alice decides to require authorization for deposits.
+            env(fset (alice, asfDepositAuth));
+            env.close();
+
+            // Even though alice is authorized to pay becky, becky is not
+            // authorized to pay alice.
+            env (pay (becky, alice, XRP (100)), ter (tecNO_PERMISSION));
+            env (pay (becky, alice, USD (100)), ter (tecNO_PERMISSION));
+            env.close();
+
+            // becky unauthorizes carol.  Should have no impact on alice.
+            env (deposit::unauth (becky, carol));
+            env.close();
+
+            env (pay (alice, becky, XRP (100)));
+            env (pay (alice, becky, USD (100)));
+            env.close();
+
+            // becky unauthorizes alice.  alice now can't pay becky.
+            env (deposit::unauth (becky, alice));
+            env.close();
+
+            env (pay (alice, becky, XRP (100)), ter (tecNO_PERMISSION));
+            env (pay (alice, becky, USD (100)), ter (tecNO_PERMISSION));
+            env.close();
+
+            // becky decides to remove authorization for deposits.  Now
+            // alice can pay becky again.
+            env(fclear (becky, asfDepositAuth));
+            env.close();
+
+            env (pay (alice, becky, XRP (100)));
+            env (pay (alice, becky, USD (100)));
+            env.close();
+        }
+    }
+
+    void run() override
+    {
+        testEnable();
+        testInvalid();
+        testPayment (jtx::supported_amendments() - featureDepositPreauth);
+        testPayment (jtx::supported_amendments() | featureDepositPreauth);
```

Comment:
No need to explicitly turn on the feature, it's included in `supported_amendments`;